src/handlers/bbb/api.js

```diff
@@ -0,0 +1,122 @@
+        })
+        .then(() => h.redirect('/meet/admin?create_success=1'))
+        .catch(async (e) => {
+            console.error('Error while creating room:', e);
+            return h.redirect(`/meet/admin?create_error=${encodeURIComponent(e.code + ' :: ' + e.message)}`);
+        });
+}
+
+const join_translations = {
+    de: {
+        not_found: 'Der gewünschte Raum existiert nicht.',
+        server_error:
+            'Beim Beitreten ist leider ein unerwarter Serverfehler aufgetreten. Bitte versuche es später erneut.',
+        admin_only: 'Du darfst diesem Raum nicht beitreten.',
+    },
+    en: {
+        not_found: 'The requested room does not exist.',
+        server_error:
+            'Unfortunately, an unexpected error occurred while trying to join the requested room. Please try again later.',
+        admin_only: 'You cannot join this room.',
+    },
+};
+async function joinRoom(request, h) {
+    const is_admin = request.state.bbb_admin_token === config.admin_token;
+
+    const { lang, slug } = request.params;
+
+    const meeting = await request.server.methods.knex('bbb').select().where({ slug }).first();
+    if (!meeting) return h.view('render-text', { text: join_translations[lang].not_found }).code(404);
+
+    if (!is_admin && meeting.admin_only)
+        return h.view('render-text', { text: join_translations[lang].admin_only }).code(404);
+
+    const params = new URLSearchParams(meeting.params);
+
+    const name = request.query.name;
+    if (!name) return h.view(`bbb-enter-name.${lang}.html`, { room_name: params.get('name') });
+
+    const is_meeting_running = await bbb('isMeetingRunning', { meetingID: slug }).then((r) => r.running);
+
+    if (!is_meeting_running && !is_admin && !meeting.anonymous_users_can_start)
+        return h.view(`bbb-wait-for-start.${lang}.html`, { room_name: params.get('name') }).header('refresh', '60');
+
+    const bbb_meeting = await bbb('getMeetingInfo', { meetingID: slug })
+        .catch(() => bbb('create', meeting.params))
+        .catch((err) => {
+            console.error('Failed to create BBB meeting:', err);
+            return h.view('render-text', { text: join_translations[lang].server_error }).code(500);
+        });
+
+    const join_url = await bbb('join', {
+        fullName: name,
+        meetingID: bbb_meeting.meetingID,
+        // This is quite silly. These passwords actually serve no security purpose. Knowing them gets you nothing since
+        // you need to know the API key to generate the checksum ("signature").
+        //
+        // The BBB devs appear to have realized this as well since according to the API docs
+        // (https://docs.bigbluebutton.org/development/api/#get-join), the `password` parameter is deprecated when
+        // joining and you're instead just supposed to use `role=MODERATOR` or `role=VIEWER`. However, for some reason
+        // this doesn't appear to actually work (might be caused by Senfcall's proxy?).
```

Comment:
I think the docs are just plain wrong. Their own test server implementation does not comply to the documentation: https://mconf.github.io/api-mate/